### PR TITLE
Specs for the shielded transfer feature in web client

### PR DESCRIPTION
**Pull request by memasdeligeorgakis**
* Added initial specs for the shielded transfer feature in web client

### CODE REVIEW

File: src/architecture/clients/web-wallet/shielded-transfers.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+<h1> Shielded Transfers In Web Client</h1>
```
Comment (tzemanovic) [inline src/architecture/clients/web-wallet/shielded-transfers.md:1]:
```suggestion
# Shielded Transfers In Web Client
```

File: src/architecture/clients/web-wallet/shielded-transfers.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+Shielded transfers are based on [MASP](https://github.com/anoma/masp) and allows users of Anoma to performs transactions where only the recipient sender and a holder of a viewing key can see the transactions details. It is based on the specifications defined at [Shielded execution](../../ledger/shielded-execution/masp.md).
```
Comment (tzemanovic) [inline src/architecture/clients/web-wallet/shielded-transfers.md:3]:
```suggestion
Shielded transfers are based on [MASP](https://github.com/anoma/masp) and allows users of Anoma to performs transactions where only the recipient, sender and a holder of a viewing key can see the transactions details. It is based on the specifications defined at [Shielded execution](../../ledger/shielded-execution/masp.md).
```

File: src/architecture/clients/web-wallet/shielded-transfers.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+### `getMaspWeb`
+* A util to return an instance of `MaspWeb` and ensure it is initiated. It it was retrieved and initiated earlier the existing instance is returned.
+```ts
+async (): Promise<MaspWeb>
+```
+
+
+### `MaspWeb`
+* this contains the methods to perform the shielded transaction related activities.
+* the consumer should always call `init()`, to ensure it is initiated. There is a utility method called `getInitialised` that returns an instantiated instance of `MaspWeb`.
```
Comment (tzemanovic) [inline src/architecture/clients/web-wallet/shielded-transfers.md:35]: should this be?
```suggestion
* the consumer should always call `getMaspWeb()`, to ensure it is initiated. There is a utility method called `getInitialised` that returns an instantiated instance of `MaspWeb`.
```

File: src/architecture/clients/web-wallet/shielded-transfers.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+//
+// pub struct ShieldedAccount {
+//     viewing_key: String,
+//     spending_key: String,
+//     payment_address: String,
+// }
+```
+
+#### `decodeTransactionWithNextTxId`
+* Utility that decodes the fetched shielded transactiosn from the ledger and returns in format that contains the shielded transaction and the id for fetching the next one.
```
Comment (tzemanovic) [inline src/architecture/clients/web-wallet/shielded-transfers.md:78]:
```suggestion
* Utility that decodes the fetched shielded transaction from the ledger and returns in format that contains the shielded transaction and the id for fetching the next one.
```

**Comment by cwgoes** [comment]
Do we have the state model of how we will store (private) notes in the web client outlined somewhere?

**Comment by memasdeligeorgakis** [comment]
> Do we have the state model of how we will store (private) notes in the web client outlined somewhere?

I added a section to explain the current flow and persistence of data here https://github.com/anoma/specs/pull/106/files#diff-8f3393b1d65d26be95acbde3fac14d4e698698d216e1eccb95e38bc071a49326R40

**Pull request merged**